### DISCUSSION TITLE
Any way to bundle webview2 runtime to application like electron?

### DISCUSSION THREAD

**Discussion by nyrf** [tauri-apps/tauri#3283] [Q&A]
Some times, we need to  install application offline, any way to bundle tauri applciation like electron's way? thanks.

**Comment by amrbashir** [comment]
you can just ask your users to install webview2 runtime beforehand or distribute a full webview2 offline installer along your tauri app installer in a zip file.

**Reply by FabianLars** [reply]
Also, the next version will most likely provide a way to use a _fixed_ version (not part of the `next` branch yet).
imo we should do something similar for the "normal" **offline** installer too (windows 7 users would appreciate it 😅), but it's also something you should be able to do with either a custom wix fragment or even inside the binary.

**Reply by nyrf** [reply]
Thanks, yes, some people need install application offline, if we can choose bundle application like electron or use system webview2 runtime should be nice.

**Comment by ngdangtu-vn** [comment]
Yeah, bundle option is ideal for app seeking for simplicity. I don't want users to concern about the dependencies. Especially when it comes to Linux. And space is not really a problem nowaday right?

**Reply by FabianLars** [reply]
If you use appimages on linux they will include the webview and (almost) all dependencies, that's really just how they work by design. `.deb` installers (and upcoming rpm support) won't include it but rely on the system's package manager, which is also just how those formats work.

This question however was really just about Windows where it's not up to the package format to "decide" the webview2 distribution mode, but since my comment above we indeed added the "fixed runtime version" mode so that's solved too (Also Microsoft accelerated the rollout so on top of windows 11, many windows 10 installations now have it already installed too).

> And space is not really a problem nowaday right?

Don't say that too loud or you'll start a fight quicker than you'd like 😂
